### PR TITLE
Better handling of motor power distribution

### PR DESCRIPTION
**Pull request by krichardsson**
This PR updates the power distribution.
 
Previously the motor power has been capped per motor if the desired thrust exceeds the maximum possible thrust. In this PR the capping has been updated to maintain the relative power of the motors and the trust of all motors are reduced equally much. The intention is to prioritize attitude over altitude to reduce the risk of flipping. 

The power distribution code has also been refactored to make it easier to understand and test.

### CODE REVIEW

**Comment by krichardsson** [comment]
@matejkarasek I have updated the power distribution a bit and thus modified the flapper power distribution file. It all looks good (in my opinion) but I have note tested on a real Flapper. The intention is that the functionality should be identical to before, would it be possible for you to take a look?
- Kristoffer

**Comment by matejkarasek** [comment]
Thanks for the heads up, will try to test it and report back soon.

The new capping method, which prioritizes the attitude, could also be interesting for the Flapper :)

**Comment by krichardsson** [comment]
Agree, I think it can be interesting for the Flapper as well. I did not want to change anything without checking with you first :-)

File: src/modules/src/power_distribution_flapper.c [hunk 1]
```diff
@@ -142,40 +142,46 @@ bool powerDistributionTest(void)
...
+  #endif
+}
+
+void powerDistributionCap(const motors_thrust_uncapped_t* motorThrustBatCompUncapped, motors_thrust_pwm_t* motorPwm)
+{
+  #if CONFIG_POWER_DISTRIBUTION_FLAPPER_REVB
+    motorPwm->motors.m2 = limitThrust(motorThrustBatCompUncapped->motors.m2, 0, UINT16_MAX); // pitch servo
+    motorPwm->motors.m3 = limitThrust(motorThrustBatCompUncapped->motors.m3, 0, UINT16_MAX); // yaw servo
+    motorPwm->motors.m1 = limitThrust(motorThrustBatCompUncapped->motors.m1, idleThrust, UINT16_MAX); // left motor
+    motorPwm->motors.m4 = limitThrust(motorThrustBatCompUncapped->motors.m4, idleThrust, UINT16_MAX); // left motor
```
Comment (matejkarasek) [inline src/modules/src/power_distribution_flapper.c:179]: copy&paste error in the comment, should be :
// right motor

**Comment by matejkarasek** [comment]
So I tested this and all seems fine, thanks!

File: src/modules/src/power_distribution_flapper.c (hunk 1 above)
Comment (krichardsson) [reply src/modules/src/power_distribution_flapper.c]: Fixed. Thanks!

**Comment by krichardsson** [comment]
Great!
Thanks @matejkarasek !

**Review by tobbeanton — APPROVED** [review]
Looks good and something we should have done a long time ago.

**Pull request merged**